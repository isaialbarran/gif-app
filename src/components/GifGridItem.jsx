import React from 'react';

const GifGridItem = ({ img }) => {
  const { url, title } = img;

  return (
    <div className='card animate__animated animate__fadeInLeftBig'>
      <img src={url} alt={title} />
    </div>
  );
};

export default GifGridItem;
