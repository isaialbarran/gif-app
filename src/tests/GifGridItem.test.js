import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../components/GifGridItem';

describe('Pruebas en <GifGridItem', () => {
  const title = 'Add tittle';
  const url = 'https://www.google.jpg';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('debe de mostrar <GifGridItem/> correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de tener un parrafo con el title', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });

  test('debe de tener una imagen con el alt y url de los props', () => {
    const img = wrapper.find('img');
    // console.log(img.prop('src'));
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('debe de tener animate__fadeInLeftBig', () => {
    const div = wrapper.find('div');
    expect(div.hasClass('animate__fadeInLeftBig')).toBe(true);
  });
});
