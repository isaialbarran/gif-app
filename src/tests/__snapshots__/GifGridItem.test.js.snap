// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem debe de mostrar <GifGridItem/> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInLeftBig"
>
  <img
    alt="Add tittle"
    src="https://www.google.jpg"
  />
  <p>
    Add tittle
  </p>
</div>
`;
